### PR TITLE
Improve: tools/vim.txt, remove irrelevant/opinionated information, add new commands

### PR DESCRIPTION
**Pull request by Gitleptune**
## Added:

- lots of new keybinds and commands
- commands section

## Changed:

-  Reorganised sections so they're in more relevant order for someone new to vim
-  Replaced `CTRL-A` syntax to `<C-a>` to fit the vim syntax

## Removed:

- .vimrc section, as it was broken and opinionated, might confuse new users reading it
- plugins section, as it didn't explain how to install them and they are not included in base vim

### CODE REVIEW

File: tools/vim.txt [hunk 1]
```diff
@@ -2,18 +2,53 @@
...
+##############################################################################
+
+
+i                   start insert mode at cursor
+I                   insert at the beginning of the line
+a                   append after the cursor
+A                   append at the end of the line
+o                   open (append) blank line below current line
+O                   open blank line above current line
+<Esc>                 exit insert mode
```
Comment (kang8) [inline tools/vim.txt:20]: pertty format
```suggestion
<Esc>               exit insert mode
```

File: tools/vim.txt [hunk 2]
```diff
@@ -84,16 +99,25 @@ Esc                 exit insert mode
...
-ct'                 change (replace) until the ' character (can change ' for any character)
+J                   join line below to the current one and add space
+gJ                  join line below to the current one without adding space
+cc                  change an entire line
+cw                  change to the end of word
+ciw                 change whole word regardless of where you are in it
+cip                 change in paragaph (between new lines)
+cap                 change around paragaph and the paragraph itself
+ci(                 change in pair of (, works with { [ " ' too
+ca(                 change in and around pair of (, works with { [ " ' too
```
Comment (kang8) [inline tools/vim.txt:110]:
```suggestion
ci(                 change in pair of (, works with { [ " ' ` < too
ca(                 change in and around pair of (, works with { [ " ' ` < too
```

File: tools/vim.txt [hunk 3]
```diff
@@ -84,16 +99,25 @@ Esc                 exit insert mode
...
+C                   change to the end of line
```
Comment (kang8) [inline tools/vim.txt:111]:
```suggestion
C                   change to the end of line (same as c$)
```

File: tools/vim.txt [hunk 4]
```diff
@@ -120,10 +147,14 @@ x                   delete current character
...
 dd                  delete (cut) a line
+diw                 delete whole word regardless of where you are in it
+dip                 delte in paragraph
+dap                 delte around paragraph and the paragraph itself
 dt'                 delete until the next ' character on the line (replace ' by any character)
 D                   delete from cursor to end of line
 :[range]d           delete [range] lines
-
+di(                 delete in pair of (, works with { [ " ' too
+da(                 delete in and around pair of (, works with { [ " ' too
```
Comment (kang8) [inline tools/vim.txt:157]:
```suggestion
di(                 delete in pair of (, works with { [ " ' ` < too
da(                 delete in and around pair of (, works with { [ " ' ` < too
```

File: tools/vim.txt [hunk 5]
```diff
@@ -132,8 +163,13 @@ D                   delete from cursor to end of line
...
 yw                  yank word
 yy                  yank (copy) a line
-2yy                 yank 2 lines
+nyy                 yank n lines
 y$                  yank to end of line
+yiw                 yank whole word regardless of where you are in it
+yip                 yank in paragaph (between new lines)
+yap                 yank around paragaph and the paragraph itself
+yi(                 yank in pair of (, works with { [ " ' too
+ya(                 yank in and around pair of (, works with { [ " ' too
```
Comment (kang8) [inline tools/vim.txt:172]:
```suggestion
yi(                 yank in pair of (, works with { [ " ' ` < too
ya(                 yank in and around pair of (, works with { [ " ' ` < too
```

**Review by kang8 — CHANGES_REQUESTED** [review]
Awesome PR!

And I have some suggest.

**Comment by Gitleptune** [comment]
Thanks for noticing those! Feel free to squash the commits, I'm not sure how to do so.

**Comment by kang8** [comment]
@Gitleptune Actually, you can squash these commit in your local compute, and force push it.

And I don't have permission, you can wait maintainer to review your PR, then merge it.

**Pull request closed without merging**